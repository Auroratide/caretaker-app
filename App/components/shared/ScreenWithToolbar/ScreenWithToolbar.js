import React from 'react';
import { View } from 'react-native';
import Toolbar from '../../material-ui/Toolbar';

import styles from './styles';

export default function ScreenWithToolbar({ title, navigation, children }) {
  return <View style={styles.container}>
    <Toolbar
      leftElement='menu'
      onLeftElementPress={() => navigation.navigate('DrawerOpen')}
      centerElement={title}
    />
    {children}
  </View>;
}
