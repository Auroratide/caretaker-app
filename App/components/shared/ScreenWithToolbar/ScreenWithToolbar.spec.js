import React, { Component } from 'react';
import { View, Text } from 'react-native';
import renderer from 'react-test-renderer';
import ScreenWithToolbar from './ScreenWithToolbar';

describe('<ScreenWithToolbar />', () => {
  let props;
  const screenWithToolbar = (children) => renderer.create(<ScreenWithToolbar {...props}>{children}</ScreenWithToolbar>);

  beforeEach(() => {
    props = {
      title: undefined,
      navigation: undefined
    };
  });

  it('should render correctly', () => {
    props.title = 'Schedule';
    props.navigation = { navigate: () => {} };
    expect(screenWithToolbar(<SomeChild />)).toMatchSnapshot();
  });

  const SomeChild = () => {
    return <View>
      <Text>hi</Text>
    </View>
  };

});
