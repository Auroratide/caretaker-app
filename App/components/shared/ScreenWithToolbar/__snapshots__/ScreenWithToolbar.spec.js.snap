// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenWithToolbar /> should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ThemeProvider
    uiTheme={
      Object {
        "palette": Object {
          "accentColor": "#FF7575",
          "primaryColor": "#FF7575",
          "secondaryColor": "#CC5D5D",
        },
        "toolbar": Object {
          "container": Object {
            "height": 50,
          },
        },
      }
    }
  />
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      hi
    </Text>
  </View>
</View>
`;
